pkg/api/methodHandlers.go

```diff
@@ -116,9 +116,14 @@ func (h *methodHandlers) execute(ctx context.Context, tracker attribute.Tracker,
 	// get a new context with the attribute bag attached
 	ctx = attribute.NewContext(ctx, ab)
 	for _, conf := range h.configs[method] {
-		// TODO: plumb ctx through uber.manager.Execute
-		_ = ctx
-		out, err := h.mngr.Execute(conf, ab, h.eval)
+		select {
+		case <-ctx.Done():
+			return newStatusWithMessage(code.Code_DEADLINE_EXCEEDED, ctx.Err().Error())
```

Comment:
I'm not sure we want to do that. For example, what if my auth adapter takes too long but would've denied the request, and all of the other adapters I executed gave an OK result? I don't think we want to return an OK in that case, right?

There's also another problem with that's written: that it's possible for Done to be closed for a reason other than deadline exceeded (e.g. callers cancels the request).